packages/x-components/src/components/global-x-bus.vue

```diff
@@ -22,29 +18,12 @@
 
       /**
        * Handles a subscription to all the events provided in the listeners with the function that
-       * will execute the callback. Also unsubscribes on beforeDestroy.
-       *
-       * @internal
+       * will execute the callback.
        */
-      const subscription = reduce(
-        listeners as XEventListeners,
-        (subscription, eventName, callback) => {
-          subscription.add(
-            (
-              xBus.on(eventName, true) as unknown as Observable<
-                SubjectPayload<EventPayload<XEventsTypes, typeof eventName>, WireMetadata>
-              >
-            ).subscribe(({ eventPayload, metadata }) => {
-              callback(eventPayload as never, metadata);
-            })
-          );
-          return subscription;
-        },
-        new Subscription()
-      );
-
-      onBeforeUnmount(() => {
-        subscription.unsubscribe();
+      Object.entries(listeners as XEventListeners).forEach(([eventName, callback]) => {
+        xBus.on(eventName as XEvent, true).subscribe(({ eventPayload, metadata }) => {
+          callback(eventPayload as never, metadata);
```

Comment:
I'm not sure if I understand this cast🤔

Reply:
It is because `callback` payload is inferred as `never` due to TS can't know `listeners` type in build time. It is generic for any event.
I'm sure we could improve the type inference, but further on. 

![Screenshot 2024-05-20 at 3 01 29 PM](https://github.com/empathyco/x/assets/10746604/684b05f3-e11e-4f61-80ff-5c0ff2eb68f2)